### DISCUSSION TITLE
Copilot Edits - Limit Working Set 10 files

### DISCUSSION THREAD

**Discussion by saul-fernandez-gstock** [community/community#143099] [Copilot Conversations]
### Select Topic Area

Question

### Body

Reading the official documentation for the new Copilot Edits tool (https://code.visualstudio.com/docs/copilot/copilot-edits#_add-files-to-the-working-set), there is a limitation of a maximum of 10 files in the Working Set per session.

Is there any way to view the content of the files that it could not generate because we have exceeded the limit of 10, even if only from the chat?

Will this limitation be removed in the future, or will it be possible to process more files in the Working Set?

Thank you very much.

**Comment by joshbigmouse** [comment]
Hi there!

The current limitation of 10 files in the Copilot Edits Working Set can indeed be restrictive, especially if you’re working on a larger project. Unfortunately, as of now, there’s no direct way to view or access the files that were excluded from generation once the 10-file limit has been reached. This limitation is set in place due to processing constraints and to optimize Copilot’s performance during each session.

If you frequently exceed this limit, one workaround could be to manually prioritize and rotate files within the Working Set. You can temporarily remove files that don’t require immediate editing, allowing you to add other essential ones to the set.

Hope this helps, and let’s hope for an update soon!

**Reply by douglasg14b** [reply]
It kinda sucks for cases where a repository has strict guidelines of 1-file-per-fucntion or class. VS repos that just god-class everything or stuff what can amount to entire repos worth of code into single files. 

This should be context-size based, not just file based.

It's useful for small projects, or very self-contained code. But isn't very useful on projects with hundreds or thousands of files.

**Comment by MarekJavurek** [comment]
Tis is really restrictive. I have set of configuration files (Micrisoft ADB2C flows and XML definitions) and copilot is great but In my project is like 25 files and one change has inpact in all of them... :(

**Comment by DaniiG2251** [comment]
Is this ever going to change? I might switch to Cursor if this limit won't be lifted. Cursor doen't have a limit.

**Reply by jamesallan93** [reply]
copilot would be better if it had some features from codeium windsurf, the AI chat has knowledge about the whole project, no file limits.
![image](https://github.com/user-attachments/assets/244db594-699e-4a4b-b205-ca74e2f764ce)

**Comment by Wladastic** [comment]
I also believe this could be improved by addind yet another workspace layer before auto binding files by itself.
Like treating the user set files higher priority IF they fit the context and keeping the ones it set itself dynamic.
Otherwise it gets stuck in a call where it tries to edit a file but cannot add it to the files as 10 is already reached.

**Comment by timint** [comment]
I don't mind restricting if it was how many files were simultaneously processed or wait time between them. But I find the max 10 files in the working set annoying. And if some of the files are time consuming it's not processing more than a couple.

A case example is when I want to process all files in a search result. Where the repetitive task of adding new files to workset is never ending.

A solution I can think of is being able to create a Queue instead of being limited to partial worksets.

Those were my two cents. In general I absolutely love Copilot. :)

**Comment by scottc-WellSky** [comment]
I've been using Copilot Edits in Agent mode. Does anyone know what the limits are for this? Seems to align with the 10 files and 6000 lines per file, but wondering if the limits are published and if folks have come up with workarounds for very large projects.

**Comment by csvan** [comment]
I am sorry, but this is ridiculously low. For Enterprise users, there should at least be an option to raise this even if it costs extra - we work with _very_ large codebases (files in the thousands per module, at least)  where 10 files is way below what is needed.

**Comment by lkraav** [comment]
So this is why I've lost edits in Agent mode, after it reached 10 edits and I thought I could just let it continue working through the challenge?

Weird part is, Agent itself or IntelliJ IDE doesn't say anything about this limit.

Agent continues to happily claim it has edited files, but in reality no further files get changed until you clear 10 edits already collected.

Quite sub-optimal UX?

EDIT also filed https://github.com/microsoft/copilot-intellij-feedback/issues/242 now.

**Reply by ebndev** [reply]
Hi @lkraav, thanks for opening that issue in the `microsoft/intellij-feedback` repo. 

(FYI for others on the thread:) From this [comment](https://github.com/microsoft/copilot-intellij-feedback/issues/242#issuecomment-2915022437), if you are using Agent mode, there should not be a limitation.

I've gone ahead and logged this issue in Edit mode as `Product Feedback` to ensure your feedback can be reviewed by the proper GitHub team. While we can't promise that every suggestion will be implemented, we want to emphasize that your feedback is instrumental in guiding our decisions and priorities.

P.S it may be helpful for someone to open an issue in the [microsoft/vscode](https://github.com/microsoft/vscode/issues) repo as well.

**Reply by lkraav** [reply]
Yeah AFAICT it has been fixed months ago.

**Comment by vhbui02** [comment]
I just add `Proceed with the next batch, and the batch after it, and the batch after it, ... untill all files are replaced` (nvm about what being replaced, you can change to anything you want), and then it keeps editing until there are no more files left to be replaced.
